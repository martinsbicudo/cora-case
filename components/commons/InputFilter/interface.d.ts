import { ChangeEvent } from 'react'

export interface InputFilterProps {
  onChange: (e: ChangeEvent<HTMLInputElement>) => void
}
