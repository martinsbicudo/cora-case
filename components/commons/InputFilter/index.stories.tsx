import { StoryObj, Meta } from '@storybook/react'

import InputFilter from '.'

export default {
  title: 'Components / InputFilter',
  component: InputFilter,
} as Meta

export const Default: StoryObj = {
  args: {},
}
