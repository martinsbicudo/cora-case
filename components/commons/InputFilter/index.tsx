import { ReactComponent as SearchIcon } from '@/public/icons/search.svg'

import { InputFilterProps } from './interface'
import * as Styled from './styles'

const InputFilter = ({ onChange }: InputFilterProps) => {
  return (
    <Styled.InputFilterWrapper>
      <Styled.InputFilterIconBox>
        <SearchIcon />
      </Styled.InputFilterIconBox>
      <Styled.InputFilter
        id="search"
        name="search"
        type="text"
        onChange={onChange}
        placeholder="Pesquisar"
      />
    </Styled.InputFilterWrapper>
  )
}

export default InputFilter
