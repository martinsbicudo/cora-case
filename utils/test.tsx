import type { ReactElement, ReactNode } from 'react'

import { ThemeProvider } from 'styled-components'

import theme from '@/styles/theme'
import { render, RenderOptions } from '@testing-library/react'

const Providers = ({ children }: { children: ReactNode }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: Providers, ...options })

export * from '@testing-library/react'
export { customRender as render }
