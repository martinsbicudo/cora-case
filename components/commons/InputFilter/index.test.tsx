import { render, screen, fireEvent } from '@/test/helpers'

import InputFilter from '.'

describe('Components / InputFilter', () => {
  const spyOnChange = jest.fn()
  const renderComponent = () => render(<InputFilter onChange={spyOnChange} />)

  it('should render component', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })

  it('should call on change prop', () => {
    renderComponent()

    const input = screen.getByRole('textbox')
    fireEvent.change(input, { target: { value: 'test' } })
    expect(spyOnChange).toHaveBeenCalledTimes(1)
    spyOnChange.mockClear()
  })
})
