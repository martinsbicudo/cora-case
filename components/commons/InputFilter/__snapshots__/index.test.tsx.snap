// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components / InputFilter should render component 1`] = `
<div>
  <fieldset
    class="sc-bcXHqe gUXnDR"
  >
    <div
      class="sc-gswNZR kfjtjk"
    >
      <i>
        icon-mock
      </i>
    </div>
    <input
      class="sc-dkrFOg hEEBVS"
      id="search"
      name="search"
      placeholder="Pesquisar"
      type="text"
    />
  </fieldset>
</div>
`;
